src/Domains/Subscription/Prices/DataTransferObject/Price.php

```diff
@@ -0,0 +1,46 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Kanvas\Subscription\Prices\DataTransferObject;
+
+use Baka\Contracts\AppInterface;
+use Baka\Contracts\CompanyInterface;
+use Baka\Users\Contracts\UserInterface;
+use Kanvas\Apps\Models\Apps;
+use Kanvas\Companies\Models\Companies;
+use Spatie\LaravelData\Data;
+
+class Price extends Data
+{
+    public function __construct(
+        public CompanyInterface $company,
+        public AppInterface $app,
+        public UserInterface $user,
+        public int $apps_plans_id,
+        public string $stripe_id,
+        public float $amount,
+        public string $currency,
+        public string $interval,
+        public ?bool $is_default = false
+    ) {
+    }
+
+    /**
+     * Create a new Price DTO from request data.
+     */
+    public static function viaRequest(array $request, UserInterface $user): self
```

Comment:
@Keysie27 pass company as param

Reply:
same for app send it as param

Reply:
There is no need to include a company when creating/updating prices, as prices and plans are associated only with apps. Companies utilize prices and plans through subscriptions.